### PR TITLE
[state-sync-v2] change TxnOutputListWithProof to TxnAndOutputListWith…

### PR DESCRIPTION
**Pull request by lightmark**
…Proof

## Motivation

TL;DR is that for the state sync v2 MVP, we'll assume that node syncing using outputs will also fetch and store (but not execute) transactions locally. We also won't backfill transaction outputs when storage is updated. Reasons for this decision include: (i) simpler programming model; (ii) easier to reason about; and (iii) less work required for executor/storage changes. 

This diff just modify the data structure from `TxnOutputListWithProof` to `TxnAndOutputListWithProof`, where the name changing is self-explanatory.

### Have you read the [Contributing Guidelines on pull requests]
Y

(Write your answer here.)

## Test Plan

Change corresponding uts.

### CODE REVIEW

**Comment by lightmark** [comment]
@JoshLind I am not also a fan of changing name. Just thought it'd better to be more clear. I will change it back.
